src/openrct2/drawing/X8DrawingEngine.cpp

```diff
@@ -47,7 +49,15 @@ void X8RainDrawer::SetDPI(rct_drawpixelinfo* dpi)
 
 void X8RainDrawer::Draw(int32_t x, int32_t y, int32_t width, int32_t height, int32_t xStart, int32_t yStart)
 {
-    const uint8_t* pattern = RainPattern;
+    const uint8_t* pattern;
+    if (climate_is_snowing())
```

Comment:
This should have nothing to do with the rain drawing code, best to move all out into its own class.